snuba/query/parser/functions.py

```diff
@@ -124,6 +132,26 @@ def parse_function(
         )
 
     name, args, alias = function_tuple
+
+    # If the first argument is a simple column name that refers to an array column
+    # (and we are not arrayJoining on that column, which would make it scalar again)
+    # we assume that the user actually means to check if any (or all) items in the
+    # array match the predicate, so we return an `any(x == value for x in array_column)`
+    # type expression. We assume that operators looking for a specific value (IN, =, LIKE)
+    # are looking for rows where any array value matches, and exclusionary operators
+    # (NOT IN, NOT LIKE, !=) are looking for rows where all elements match (eg. all NOT LIKE 'foo').
+    if name in FUNCTION_TO_OPERATOR:
+        if len(args) == 2 and args[0] in dataset_columns:
```

Comment:
please call out the limitation that this only works if the first argument is a bare column and not a function returning a column (like `assumeNotNull(stacktraces.....)`)